demos/smart_classroom_demo_gapi/main.cpp

```diff
@@ -0,0 +1,221 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <string>
+#include "person_action_detector.hpp"
+#if defined(HAVE_OPENCV_GAPI)
+#include <samples/ocv_common.hpp>
+#include <opencv2/gapi.hpp>
+#include <opencv2/gapi/core.hpp>
+#include <opencv2/gapi/imgproc.hpp>
+#include <opencv2/gapi/fluid/core.hpp>
+#include <opencv2/gapi/infer.hpp>
+#include <opencv2/gapi/infer/ie.hpp>
+#include <opencv2/gapi/cpu/gcpukernel.hpp>
+#include <opencv2/gapi/streaming/cap.hpp>
+
+std::string getBinPath(const std::string & pathXML) {
+    std::string pathBIN(pathXML); 
+    return pathBIN.replace(pathBIN.size() - 3, 3 , "bin");
+}
+namespace show {
+    const std::string actions[] = {
+        "sitting", "standing", "rising hand"
+    };
+
+    void DrawResults(cv::Mat &frame, const std::vector<cv::Rect> &faces) {
+        for (auto it = faces.begin(); it != faces.end(); ++it) {
+            const auto &rc = *it;
+            cv::rectangle(frame, rc, {200, 200, 200},  2);
+        }
+    }
+
+    void DrawResults(cv::Mat &frame, const std::vector<Detections> &persons) {
+        for (auto it = persons.begin(); it != persons.end(); ++it) {
+            const auto &rc = (*it).rect;
+            cv::rectangle(frame, rc, {200, 200, 200},  2);
+            std::stringstream ss;
+            ss << actions[(*it).action_label];
+            cv::putText(frame, ss.str(),
+                        cv::Point(rc.x, rc.y - 15),
+                        cv::FONT_HERSHEY_COMPLEX_SMALL,
+                        1,
+                        cv::Scalar(0, 0, 255));
+        }
+    }
+} // namespace show
+
+namespace {
+    const std::string about =
+        "classroom";    
+    const std::string keys =
+    "{ h help |   | print this help message }"
+    "{ input  |   | Path to an input video file }"
+    "{ fdm    |   | IE face detection model IR }"
+    "{ pam    |   | IE person detection action recognition model IR }" 
+    "{ fdd    |   | IE face detection device }";
+}
+
+namespace custom {
```

Comment:
Can i move this (kernels, nets) to the appropriate files from main.cpp?

Reply:
Yes you can, but do you actually need this?